packages/apps-sdk/src/sdk.ts

```diff
@@ -31,9 +34,21 @@ export class SDK {
    */
   private configureApp(): void {
     // First we obtaing the id of this app from the url
-    const parsed = queryString.parse(window.location.search);
+    const parsed = qs.parse(window.location.search.substring(1), { decoder });
     const parsedAppName = Array.isArray(parsed.appName) ? parsed.appName[0] : parsed.appName;
+    const parsedQueryParams = Array.isArray(parsed.queryParams) ? parsed.queryParams[0] : parsed.queryParams;
+    const parsedInitialData = Array.isArray(parsed.initialData) ? parsed.initialData[0] : parsed.initialData;
+
     this.appName = parsedAppName ? parsedAppName : 'messenger-iframe';
+    this.queryParams = parsedQueryParams;
+    this.initialData = parsedInitialData;
+
+    const mode = this.queryParams && this.queryParams.mode ? this.queryParams.mode : APP_MODE_IFRAME;
+    const channgelConfig = mode === APP_MODE_POPUP ? { window: window.opener } : { window: window.parent };
```

Comment:
chann`g`elConfig ?

Reply:
channelConfig is an object that allow the channelManager (post-robot using postMessage now) to be configured.
Depending on the app mode (iframe or popup) we need different configuration. The window object we need to talk to changes, in one case we need to talk with window.parent other with window.opener